andela/mmdp_site
# MMDP-134 Authorize access based on group permissions

**What does this PR do?**

Group members can access the application based on defined permissions.

**Description of Task to be completed?**

1. Remove existing isAdmin field on users model
2. Create Login, Logout testing helpers.
3. Ensure that groups that have a dependant cannot be deleted
4. Integrate authorization for groups resource.
5. Refactor groups tests.
6. Repeat step 4 and 5 for all resources.

**How should this be manually tested?**

Pull and checkout to the `ft-group-based-authorization-MMDP-134`. Try making API calls to `user`, `group`, `cms -all` crud. By default they should fail. To access them you will need to log in use the token as the authorization header. The logged in user will need to have the permission required for each of the action. Lack of this will result in a `403`.

**Any background context you want to provide?**

Any pertinent information that should be considered

**What are the relevant Jira issues?**

[MMDP-134](https://viisaus.atlassian.net/browse/MMDP-134)

**Screenshots (if appropriate)**
Missing authorization header

<img width="926" alt="screenshot 2019-01-31 at 11 28 23" src="https://user-images.githubusercontent.com/8180548/52041537-c0dd1500-254b-11e9-89cd-2a80d639f2f4.png">

Authenticated but not allowed to perform action
<img width="924" alt="screenshot 2019-01-31 at 11 25 25" src="https://user-images.githubusercontent.com/8180548/52041328-1f55c380-254b-11e9-82a4-f1af3d3a5ea0.png">



**Questions:**

N/A
